### ISSUE TITLE
usable output

### ISSUE THREAD

**Issue by ConnyOnny** [pornel/duplicate-kriller#4]
I'd like to use this for _finding_ duplicates and then deciding myself what to do with them. I'd like to write a script that runs this program and for every duplicate asks me what to do with it (e.g. make a hardlink or delete one of the files or whatever). Parsing the output of your tool is quite inconvenient.

    Dry run. No files will be changed.
    0+0 dupes. 1429+1 files scanned. /home/connyonny/…
    Found dupe /home/connyonny/{doc.txt => doc_2016}

Could you please add a more easily parsable output format? I'd vote for JSON because there are JSON parsers for every language and you can easily add information in the future without breaking programs using the data.

**Example:**

```JSON
{"duplicates":[
    {"paths":[
        "/home/connyonny/doc.txt",
        "/home/connyonny/doc_2016",
        "/home/connyonny/backupdocs/doc_2016"
    ],
    "sha512":"sdfiuhsidfiazsgfidfsdff",
    "mimetype":"text/plain"},
    {"paths":[
        "/tmp/manual.pdf",
        "/home/connyonny/Downloads/manual.pdf"
    ],
    "sha512":"idfgsodfhodfhsdufhisduh",
    "mimetype":"application/pdf"}
],
"creator":"duplicate-kriller 1.1",
"scan-time":"2541ms",
"files-scanned":1429
}
```

The `sha512`, `mimetype` and `scan-time` fields are just examples for what could be added at some point in the future.

**Comment by pornel** [comment]
Would you like to try adding this?

The duplicates can be read from the scanner using this method:

https://github.com/pornel/duplicate-kriller/blob/07a7b9e9d2eaf64cf930af0ad4ebf50cc27f3a82/src/dupe.rs#L212

and it could be done after the main loop:

https://github.com/pornel/duplicate-kriller/blob/07a7b9e9d2eaf64cf930af0ad4ebf50cc27f3a82/src/main.rs#L39

If you'd like to suppress regular output, you could either just add a setting that skips the UI calls:

https://github.com/pornel/duplicate-kriller/blob/07a7b9e9d2eaf64cf930af0ad4ebf50cc27f3a82/src/dupe.rs#L100

or go for a fancier option of making the `UI` type a trait, and implementing a no-op UI variant:

https://github.com/pornel/duplicate-kriller/blob/07a7b9e9d2eaf64cf930af0ad4ebf50cc27f3a82/src/ui.rs

**Comment by ConnyOnny** [comment]
closed by #6

**Comment by pornel** [comment]
Thanks for great work. I've now released a new version.
